src/Source/FileSource.js

```diff
@@ -76,74 +75,55 @@ function fileParser(text) {
  *     extrude: () => 5000,
  * });
  *
- * // Create and append the source
- * ariege.source = new itowns.FileSource({
- *     url: 'https://raw.githubusercontent.com/gregoiredavid/france-geojson/master/departements/09-ariege/departement-09-ariege.geojson',
- *     projection: 'EPSG:4326',
- *     format: 'application/json',
- *     zoom: { min: 7, max: 7 },
- * });
- *
- * view.addLayer(ariege);
- *
- * @example
- * // Create the source
- * const gpxSource = new itowns.FileSource({
- *     url: 'https://raw.githubusercontent.com/iTowns/iTowns2-sample-data/master/ULTRA2009.gpx',
- *     projection: 'EPSG:4326',
- * });
- *
- * // Create the layer
- * const gpxLayer = new itowns.ColorLayer('Gpx', {
- *     name: 'Ultra 2009',
- *     transparent: true,
- *     source: gpxSource,
- * });
+ * itowns.Fetcher.json('https://raw.githubusercontent.com/gregoiredavid/france-geojson/master/departements/09-ariege/departement-09-ariege.geojson')
+ *     .then(function _(geojson) {
+ *         return itowns.GeoJsonParser.parse(geojson, {
+ *             buildExtent: true,
+ *             crsIn: 'EPSG:4326',
+ *             crsOut: view.tileLayer.extent.crs(),
+ *             mergeFeatures: true,
+ *             withNormal: false,
+ *             withAltitude: false,
+ *         });
+ *     }).then(function _(parsedData) {
+ *         ariege.source = new itowns.FileSource({
+ *             projection: 'EPSG:4326',
+ *             parsedData,
+ *         });
  *
- * // Add the layer
- * view.addLayer(gpxLayer);
+ *         return view.addLayer(ariegeLayer);
+ *     });
  */
 class FileSource extends Source {
     /**
      * @param {Object} source - An object that can contain all properties of a
-     * FileSource and {@link Source}. Only <code>url</code> and
-     * <code>projection</code> are mandatory.
+     * FileSource and {@link Source}. Only <code>projection</code> is mandatory.
      * @param {string} crsOut - The projection of the output data after parsing.
      *
      * @constructor
      */
-    constructor(source, crsOut) {
+    constructor(source) {
         if (!source.projection) {
             throw new Error('source.projection is required in FileSource');
         }
+
+        // the fake url is for when we use the fetchedData or parsedData mode
+        source.url = source.url || 'fake-file-url';
```

Comment:
could you verify if `source.url` is undefined so `fetchedData` or `parsedData` are defined?

Reply:
Yes good idea

Reply:
Done